Nix CI failed since irb 1.8.0 required rdoc gem as a dependency

Nix CI failed since irb 1.8.0 required rdoc gem as a dependency

https://github.com/ruby/irb/pull/648
https://github.com/kachick/rspec-matchers-power_assert_matchers/actions/runs/6032140237/job/16366864761#step:7:161

```
An error occurred while installing psych (5.1.0), and Bundler cannot continue.

In Gemfile:
  rspec-matchers-power_assert_matchers was resolved to 0.2.0, which depends on
    irb was resolved to 1.8.0, which depends on
      rdoc was resolved to 6.5.0, which depends on
        psych
Error: Process completed with exit code 5.
```

--- a/flake.nix
+++ b/flake.nix
@@ -28,6 +28,11 @@
               bashInteractive
 
               ruby
+              # Required to build psych via irb dependency
+              # https://github.com/kachick/rspec-matchers-power_assert_matchers/issues/122
+              # https://github.com/ruby/irb/pull/648
+              libyaml
+
               dprint
               tree
               nil

